expr/literals.go

```diff
@@ -144,7 +154,25 @@ type PrimitiveLiteral[T PrimitiveLiteralValue] struct {
 
 func (*PrimitiveLiteral[T]) isRootRef() {}
 func (t *PrimitiveLiteral[T]) String() string {
-	return fmt.Sprintf("%s(%v)", t.Type, t.Value)
+	return fmt.Sprintf("%s(%s)", t.Type.String(), t.ValueString())
+}
+func (t *PrimitiveLiteral[T]) ValueString() string {
+	switch x := any(t.Value).(type) {
+	case types.Date:
+		d := civil.Date{Year: 1970, Month: time.January, Day: 1}
+		d = d.AddDays(int(x))
+		return d.String()
+	case types.Time:
+		t := time.UnixMicro(int64(x))
+		return t.UTC().Format(time.TimeOnly)
+	case types.Timestamp:
+		t := time.UnixMicro(int64(x))
+		return t.UTC().Format(time.RFC3339)
+	case types.TimestampTz:
+		t := time.UnixMicro(int64(x))
+		return t.UTC().Format(time.RFC3339)
+	}
```

Comment:
I wonder if we should add `String()` methods to the `types.Date`/`types.Time`/etc. types directly which then implements these conversions rather than having to perform the type switch here.

Reply:
I do agree we do need to move this calculation out and the types would be one such place.  But that would mean passing the literal internals to the type class for conversion.  I'd rather see us have literal implementations for all types (perhaps loose wrappers around the existing generics) to hang that calculation on.

Reply:
I'm confused, what literal internals would be being passed here? I'm proposing being able to do `return t.Value.String()` 

We could add `String()` to the Type interface, and then implement it for the types, or perhaps just a subset of the types and this could be: 

```go
if s, ok := t.Value.(fmt.Stringer); ok {
     return s.String()
}

return fmt.Sprintf("%v", t.Value)
```

Reply:
Another alternative might be adding a `ToTime` method on these. For example see https://github.com/apache/iceberg-go/blob/af5a9908a83444ca13b19d24a27200bc7323a2b6/types.go#L519

Reply:
t.Value is an int64 for the time fields, the type is t.Type.  An approach putting the ToTime on the type would result in something like `t.Type.ToTime(t.Value)`.  Note that Iceberg's types are the equivalent of our literals which is where I'd rather see the method.  I suppose we could create a new TimeLiteral interface and add a generic ToTime method but we'd still have the type case switch.

Reply:
Well, iceberg has literals also, and you can see the way the ToTime is used: https://github.com/apache/iceberg-go/blob/af5a9908a83444ca13b19d24a27200bc7323a2b6/literals.go#L642

Though those literals aren't fully generic like substrait's here. 

That said, if we had the TimeLiteral interface, then we wouldn't need the type switch, it could just be the type assertion example: 

```go
if tm, ok := t.Value.(TimeLiteral); ok {
         return tm.Format(....)
}

return fmt.Sprintf(...)
```

Which now that I see it, I rather like it lol

Reply:
Keep in mind, I'm suggesting to put the ToTime on `types.Date`, not `types.DateType`

Reply:
done